core/src/main/java/org/bitcoinj/crypto/DeterministicKey.java

```diff
@@ -160,25 +145,37 @@ public DeterministicKey(List<ChildNumber> childNumberPath,
                             @Nullable DeterministicKey parent,
                             int depth,
                             int parentFingerprint) {
-        super(priv, new LazyECPoint(ECKey.publicPointFromPrivate(priv), true));
-        checkArgument(chainCode.length == 32);
-        this.parent = parent;
-        this.childNumberPath = HDPath.M(Objects.requireNonNull(childNumberPath));
-        this.chainCode = Arrays.copyOf(chainCode, chainCode.length);
-        this.depth = depth;
-        this.parentFingerprint = ascertainParentFingerprint(parent, parentFingerprint);
+        this(priv, new LazyECPoint(ECKey.publicPointFromPrivate(priv), true), depth, parent,
+                ascertainParentFingerprint(parent, parentFingerprint), chainCode,
+                HDPath.M(childNumberPath));
     }
 
-    
     /** Clones the key */
     public DeterministicKey(DeterministicKey keyToClone, DeterministicKey newParent) {
-        super(keyToClone.priv, keyToClone.pub);
-        this.parent = newParent;
-        this.childNumberPath = keyToClone.childNumberPath;
-        this.chainCode = keyToClone.chainCode;
+        this(keyToClone.priv, keyToClone.pub, keyToClone.childNumberPath.size(), newParent,
+                newParent.getFingerprint(), keyToClone.chainCode, keyToClone.childNumberPath);
         this.encryptedPrivateKey = keyToClone.encryptedPrivateKey;
-        this.depth = this.childNumberPath.size();
-        this.parentFingerprint = this.parent.getFingerprint();
+    }
+
+    /**
+     * Canonical constructor.
+     *
+     * @param priv            private key, or {@code null} if public key only
+     * @param pub             public key, corresponding to private key (if present)
+     * @param depth           depth of this key in the path, {@code 0} means master key
+     * @param parent          parent deterministic key, or {@code null} if unknown or master key
+     * @param chainCode       32 bytes of chain code
+     * @param childNumberPath path leading up to this key
+     */
+    private DeterministicKey(@Nullable BigInteger priv, LazyECPoint pub, int depth, @Nullable DeterministicKey parent,
+                             int parentFingerprint, byte[] chainCode, HDPath childNumberPath) {
```

Comment:
I would actually name the `HDPath` variable `hdPath` or something similar.

Reply:
Addressed, but for now only in the new constructor.